### PR TITLE
perf: 公招三星优先tag 改为复选框

### PR DESCRIPTION
**Pull request by AlezHibali**
## Description

This PR replaces the existing `AutoRecruitHighPriority (公招三星优先tag)` textbox with a combination of `CheckComboBox` and `ListBox` for improved user interaction and better tag selection functionality.

### Changes:
- Replaced the `AutoRecruitHighPriority` textbox with a `CheckComboBox` to allow multi-tag selection.
- Implemented a `ListBox` to display the selected tags.
- Updated the data binding to ensure proper handling of selected tags.
- Refactored related event handling logic to integrate with the new components.
- Reword the tips and title for `AutoRecruitHighPriority`.

## Motivation

The previous implementation used a simple `TextBox` for selecting priority tags, which was not user-friendly, especially for selecting multiple tags. **Nobody wants to type those tags in string with no typo at all. 太不人性化了:(** 
This update improves usability by introducing a `CheckComboBox` for better tag selection and a `ListBox` to display selected tags, enhancing overall functionality.

## How Has This Been Tested?

- Manually tested the `CheckComboBox` and `ListBox` functionalities to ensure proper tag selection and display.
- Verified data binding for the selected tags in the UI.
- Functionality is good by manual testing.
- Other functionalities are unchanged and testing in Visual Studio **does not give any error**.

## Screenshots for Tested Functionality
<p align="center">
  Tag selection
  Tag selected in Game
</p>

## Screenshots for Changes

| Before | After |
|-------|-------|
<p align="center">
  Before
  After
</p>

### CODE REVIEW

**Comment by status102** [comment]
其他语言文件也改改

**Comment by AlezHibali** [comment]
> 其他语言文件也改改

`src\MaaWpfGui\Res\Localizations` 里的5个xaml都改好了

File: src/MaaWpfGui/Views/UserControl/AutoRecruitSettingsUserControl.xaml [hunk 1]
```diff
@@ -41,12 +41,19 @@
                 SelectedValue="{Binding SelectExtraTags}"
                 SelectedValuePath="Value"
                 Style="{StaticResource ComboBoxExtend}" />
-            <TextBox
+            <controls:TextBlock 
+                Text="{DynamicResource AutoRecruitHighPriority}" 
+                TextWrapping="Wrap" 
+                HorizontalAlignment="Left"
+                Margin="10,0,0,0" />
```
Comment (status102) [inline src/MaaWpfGui/Views/UserControl/AutoRecruitSettingsUserControl.xaml:48]:
```suggestion
```

File: src/MaaWpfGui/Views/UserControl/AutoRecruitSettingsUserControl.xaml [hunk 2]
```diff
@@ -41,12 +41,19 @@
...
                 Margin="0,10"
                 hc:InfoElement.Title="{DynamicResource AutoRecruitHighPriority}"
-                Style="{StaticResource TextBoxExtend}"
-                Text="{Binding AutoRecruitFirstList}"
-                ToolTip="{DynamicResource AutoRecruitHighPriorityTooltip}" />
+                ItemsSource="{Binding AutoRecruitTagList}"
+                hc:ListBoxAttach.SelectedItems="{Binding AutoRecruitFirstList}"
+                ToolTip="{DynamicResource AutoRecruitHighPriorityTooltip}"
+                SelectionMode="Multiple"
+                ScrollViewer.VerticalScrollBarVisibility="Visible" />
```
Comment (status102) [inline src/MaaWpfGui/Views/UserControl/AutoRecruitSettingsUserControl.xaml:56]:
```suggestion
                hc:ListBoxAttach.SelectedItems="{Binding AutoRecruitFirstList}"
                ItemsSource="{Binding AutoRecruitTagList}"
                SelectionMode="Multiple"
                Style="{StaticResource CheckComboBoxExtend}"
                ToolTip="{DynamicResource AutoRecruitHighPriorityTooltip}" />
```

File: src/MaaWpfGui/ViewModels/UI/SettingsViewModel.cs [hunk 3]
```diff
@@ -3416,18 +3416,49 @@ public bool CustomStageCode
...
+        private List<string> _autoRecruitFirstAllList = new(_autoRecruitOptionsDict.Keys);
+
+        public List<string> AutoRecruitTagList
+        {
+            get => _autoRecruitFirstAllList;
+            set => SetAndNotify(ref _autoRecruitFirstAllList, value);
+        }
+
+        private object[] _autoRecruitFirstList = ConfigurationHelper
+            .GetValue(ConfigurationKeys.AutoRecruitFirstList, "A B C")
```
Comment (status102) [inline src/MaaWpfGui/ViewModels/UI/SettingsViewModel.cs:3448]: 默认值得改改

File: src/MaaWpfGui/Res/Localizations/en-us.xaml [hunk 4]
```diff
@@ -793,8 +793,8 @@ The video aspect ratio needs to be 16:9 without interference factors such as bla
     <system:String x:Key="DefaultNoExtraTags">Additional Tags are not selected by default</system:String>
     <system:String x:Key="SelectExtraTags">If higher ★ tags appear, always select 3 tags</system:String>
     <system:String x:Key="SelectExtraOnlyRareTags">Pick as many high-star Tags as possible</system:String>
-    <system:String x:Key="AutoRecruitHighPriority">Preference for 3★ Tags (split by semicolon)</system:String>
-    <system:String x:Key="AutoRecruitHighPriorityTooltip">Substring is sufficient, will select as many Tag preferences as possible</system:String>
+    <system:String x:Key="AutoRecruitHighPriority">Preference for 3★ Tags</system:String>
+    <system:String x:Key="AutoRecruitHighPriorityTooltip">Will select as many Tag preferences as possible</system:String>
```
Comment (Constrat) [inline src/MaaWpfGui/Res/Localizations/en-us.xaml:797]: lgtm

**Review by Constrat — APPROVED** [review]

**Comment by status102** [comment]
词条的i18n也麻烦加加

**Comment by status102** [comment]
> 试了一下，这个输入框也必须输入对应服务器语言的tag才能生效，这下要折腾了
> 
> 当前主线版本的大概的操作逻辑是：
> 
> 1. 执行任务时直接传入使用对应语言的tag优先列表
> 2. ocr识别界面语言，通过对应服务器的 task.json 的 RecruitTags ocrReplace 替换成中文tag
> 3. 再通过对应服务器的 recruitment.json 转换为对应语言的tag
> 4. 然后再查找有没有优先的tag

我看的跟你讲的有点出入，只有输出给gui时候进行语言转换，内部处理是全中文。不过不确定看的是不是对的
@Constrat

**Comment by status102** [comment]
一大坨（睡了，你们测测

**Comment by AlezHibali** [comment]
> 一大坨（睡了，你们测测

辛苦辛苦， 我在官服测试过了 不管maa gui是什么语言都可以正常function. 但是我没有测其他服务器的（没有国际服的号:/）

**Comment by AlezHibali** [comment]
> 词条的i18n也麻烦加加

这个你加过了吗，还是指在 https://weblate.maa-org.net/ 这里加

**Comment by status102** [comment]
> Warning
> 
> 存在未确认的链接，请谨慎访问。 There are unconfirmed links, please visit with caution.
> 
> Warning
> 
> 存在未确认的链接，请谨慎访问。 There are unconfirmed links, please visit with caution.
> 
> > 词条的i18n也麻烦加加
> 
> 这个你加过了吗？ 还是指在 [weblate.maa-org.net](https://weblate.maa-org.net/) 这里加?

https://github.com/MaaAssistantArknights/MaaAssistantArknights/commit/a0d76c1d08c5ee3d8bba369584b23e7b085f9e07

就是这个

**Comment by AlezHibali** [comment]
噢噢噢

**Comment by AlezHibali** [comment]
这个PR 还有什么需要改的吗

**Comment by Manicsteiner** [comment]
我这里测试没问题了👍

**Pull request merged**